modules/fedimint-ln-server/src/lib.rs

```diff
@@ -1036,9 +1035,31 @@ impl Lightning {
         dbtx: &mut ModuleDatabaseTransaction<'_>,
         gateway: LightningGateway,
     ) {
+        // Garbage collect expired gateways (since we're already writing to the DB)
```

Comment:
Nit: This might make guardian registration fail more often, since now if two gateways try to register at the same time, they both might try to delete the same key and the database transaction will fail. I don't think we can do this anywhere else though. Clients are already expected to retry

Reply:
Someone in the call today mentioned the idea of adding a lock. What do you think about adding an in-memory deletion lock? If the lock is free, grab it and delete expired gateways. If the lock is taken, skip deleting expired gateways.

Reply:
IMO not worth it right now, we can add later if we see failures

Reply:
Sounds good! I added a comment to the code here about this "gotcha".